buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -1011,6 +1031,27 @@ private void createWorkingDir(Path distroExtractDir) throws IOException {
      * @param destinationRoot destination to link to
      */
     private void syncWithLinks(Path sourceRoot, Path destinationRoot) {
+        sync(sourceRoot, destinationRoot, (Path d, Path s) -> {
+            try {
+                Files.createLink(d, s);
+            } catch (IOException e) {
+                // Note does not work for network drives, e.g. Vagrant
+                throw new LinkCreationException("Failed to create hard link " + d + " pointing to " + s, e);
+            }
+        });
+    }
+
+    private void syncWithCopy(Path sourceRoot, Path destinationRoot) {
+        sync(sourceRoot, destinationRoot, (Path d, Path s) -> {
+            try {
+                Files.copy(s, d);
```

Comment:
Unlike creating a symlink, this is a costly operation so how do we handle scenarios in tests where we do things like restart nodes. Since this is called in the `start()` method do we do this copy every time?

Reply:
I have to check if we can avoid that. There seems some Test expect an explicit clean working dir.

Reply:
I think we can potentially manage the state in the node itself. Basically if the backing distribution extracted dir hasn't changed, and we've already performed a copy, then skip it. I can't see why any tests would need the working directory "cleaned" as those files should never actually be modified, if they did, then shared hard links wouldn't work. We do have to check if the configured distribution changed tho, as that is mutable in the case of BWC tests the bump the distribution version.

Reply:
Files have only been copied (hard linked) if the expected directory already existed. This expected directory contains the version number so a change in the requested distribution version will trigger a redo of the linking/copying. To be clear: that has been the case before that change already and the behaviour hasn't changed with this PR